contracts/strategies/sablier-v2/LockupDynamicStrategy.sol

```diff
@@ -0,0 +1,425 @@
+                revert ALLOCATION_EXCEEDS_POOL_AMOUNT();
+            }
+
+            recipient.grantAmount = grantAmount;
+            recipient.recipientStatus = InternalRecipientStatus.Accepted;
+
+            emit RecipientStatusChanged(recipientId, InternalRecipientStatus.Accepted);
+            emit Allocated(recipientId, recipient.grantAmount, pool.token, _sender);
+        } else if (
+            recipient.recipientStatus != InternalRecipientStatus.Rejected // no need to reject twice
+                && recipientStatus == InternalRecipientStatus.Rejected
+        ) {
+            recipient.recipientStatus = InternalRecipientStatus.Rejected;
+            emit RecipientStatusChanged(recipientId, InternalRecipientStatus.Rejected);
+        }
+    }
+
+    /// @notice Distribute the upcoming milestone
+    /// @param _sender The sender of the distribution
+    function _distribute(address[] memory _recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyPoolManager(_sender)
+    {
+        uint256 recipientLength = _recipientIds.length;
+        for (uint256 i = 0; i < recipientLength;) {
+            _distributeToLockupDynamic(_recipientIds[i], _sender);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    function _distributeToLockupDynamic(address _recipientId, address _sender) private {
+        Recipient memory recipient = _recipients[_recipientId];
+        IAllo.Pool memory pool = allo.getPool(poolId);
+
+        uint128 amount = uint128(recipient.grantAmount);
+
+        LockupDynamic.CreateWithDeltas memory params = LockupDynamic.CreateWithDeltas({
+            asset: SablierIERC20(pool.token),
+            broker: broker,
+            cancelable: recipient.cancelable,
+            recipient: recipient.recipientAddress,
+            segments: recipient.segments,
+            sender: address(this),
+            totalAmount: amount
+        });
+
+        poolAmount -= amount;
+        IERC20(pool.token).forceApprove(address(lockupDynamic), amount);
+        uint256 streamId = lockupDynamic.createWithDeltas(params);
+        _recipientStreamIds[_recipientId].push(streamId);
+
+        emit Distributed(_recipientId, recipient.recipientAddress, amount, _sender);
+    }
+
+    /// @notice Get the recipient
+    /// @param _recipientId Id of the recipient
+    function _getRecipient(address _recipientId) internal view returns (Recipient memory recipient) {
+        recipient = _recipients[_recipientId];
+    }
+
+    /// @notice Check if sender is identity owner or member
+    /// @param _anchor Anchor of the identity
+    /// @param _sender The sender of the transaction
+    function _isIdentityMember(address _anchor, address _sender) internal view returns (bool) {
+        IRegistry registry = allo.getRegistry();
+        IRegistry.Identity memory identity = registry.getIdentityByAnchor(_anchor);
```

Comment:
We have updated `identity` to `profile` nomenclature.

Example:

`IRegistry.Identity` -> `IRegistry.Profile`
`isOwnerOrMemberOfIdentity` -> `isOwnerOrMemberOfProfile`

🚨 Anywhere `identity` was used previously is now `profile`